addressing/README.md

```diff
@@ -147,6 +147,33 @@ within it. Using our example above, decapsulating either `/tcp/1234/ws` _or_
 unsurprising if you consider the utility of the `/ip4/7.7.7.7/ws` address that
 would result from simply removing the `tcp` component.
 
+### The security multiaddr
+
+Peers MAY advertise their addresses without the security protocol, e.g.
+`/ip4/6.6.6.6/tcp/1234/` or `/ip4/6.6.6.6/udp/1234/quic`. The security handshake
+protocol is then negotiated using the [multistream](../multistream.md). This is
+the way the libp2p handshake worked until mid 2021.
+This poses a security problem, as the negotiation was not authenticated and
+therefore susceptible to man-in-the-middle attacks. A MITM could modify the list
+of supported handshake protocols, thereby forcing a downgrade to a (potentially)
+less secure handshake protocol. Note that since QUIC is standardized to use
+TLS 1.3, no handshake protocol needs to be negotiated when using QUIC.
+
+Peers SHOULD encapsulate the security protocol in the addresses they advertise,
```

Comment:
Nit pick: Given that this is the section about security protocols in multiaddr, wouldn't it make more sense to have this paragraph and the next go first, especially as multiaddr without security protocols will not be used for much longer once this is in place?

Reply:
Where exactly? Do you mean before Encapsulation?

Reply:
Does the following diff improve my suggestion?

```diff
diff --git a/addressing/README.md b/addressing/README.md
index abc6bc5..0967407 100644
--- a/addressing/README.md
+++ b/addressing/README.md
@@ -149,16 +149,6 @@ would result from simply removing the `tcp` component.
 
 ### The multiaddr security component
 
-Peers MAY advertise their addresses without a security protocol, e.g.
-`/ip4/6.6.6.6/tcp/1234/` or `/ip4/6.6.6.6/udp/1234/quic`. The security handshake
-protocol is then negotiated using [multistream-select](../connections/README.md#multistream-select). This is
-the way the libp2p handshake worked until mid 2021.
-This poses a security problem, as the negotiation was not authenticated and
-therefore susceptible to man-in-the-middle attacks. A MITM could modify the list
-of supported handshake protocols, thereby forcing a downgrade to a (potentially)
-less secure handshake protocol. Note that since QUIC is standardized to use
-TLS 1.3, no handshake protocol needs to be negotiated when using QUIC.
-
 Peers SHOULD encapsulate the security protocol in the addresses they advertise,
 for example `/ip4/6.6.6.6/tcp/1234/tls` for a TLS 1.3 server listening on TCP
 port 1234 and `/ip4/6.6.6.6/tcp/1235/noise` for a Noise server listening on TCP
@@ -180,6 +170,15 @@ Implementations using [Protocol Select](https://github.com/libp2p/specs/pull/349
 Note that it’s not valid to assume that any node that encapsulated the security
 protocol in their multiaddr also supports Protocol Select.
 
+Peers MAY advertise their addresses without a security protocol, e.g.
+`/ip4/6.6.6.6/tcp/1234/` or `/ip4/6.6.6.6/udp/1234/quic`. The security handshake
+protocol is then negotiated using [multistream-select](../connections/README.md#multistream-select). This is
+the way the libp2p handshake worked until mid 2021.
+This poses a security problem, as the negotiation was not authenticated and
+therefore susceptible to man-in-the-middle attacks. A MITM could modify the list
+of supported handshake protocols, thereby forcing a downgrade to a (potentially)
+less secure handshake protocol. Note that since QUIC is standardized to use
+TLS 1.3, no handshake protocol needs to be negotiated when using QUIC.
 ```